ui/permissions.py

```diff
@@ -68,4 +204,7 @@ def has_permission(self, request, view):
             uuid.UUID(collection_key)
         except ValueError:
             raise ValidationError('wrong UUID format for {}'.format(collection_key))
-        return Collection.objects.filter(owner=request.user, key=collection_key).exists()
+        collection = Collection.objects.filter(key=collection_key)
+        if collection:
+            return has_admin_permission(collection.first(), request)
+        return False
```

Comment:
The more succinct way also seems less readable/intuitive to me, also won't work quite right as is (collection is an empty list, not literally `None`, so would have to do something like `len(collection) > 0`.